src/Common/tests/System/Net/Http/LoopbackProxyServer.cs

```diff
@@ -63,27 +66,46 @@ private void Start()
         {
             Task.Run(async () =>
             {
+                var activeTasks = new ConcurrentDictionary<Task, int>();
+
                 try
                 {
                     while (true)
                     {
                         Socket s = await _listener.AcceptAsync().ConfigureAwait(false);
-                        var ignored = Task.Run(async () =>
+
+                        var connectionTask = Task.Run(async () =>
                         {
                             try
                             {
                                 await ProcessConnection(s).ConfigureAwait(false);
                             }
-                            catch (Exception)
+                            catch (Exception ex)
                             {
-                                // Ignore exceptions.
+                                Errors?.Invoke(this, new ErrorEventArgs(ex));
```

Comment:
Currently no one is registering with the event.  What's the purpose of having it if no one listening to it?

Reply:
I can get rid of it, but it's useful for troubleshooting issues that are caused by `LoopbackProxyServer` rather than your own test's code. An appropriate compromise may be to use `EventSource` logging as @davidsh suggested.

Reply:
Adding this `Event` mechanism introduces a new design pattern into our testing logic.  We  should use the existing test logging mechanism we already have with the System.Net.Test EventSource we've defined. That is how we can log and review these unexpected errors.

Reply:
Agreed and updated.